lib/device.ex

```diff
@@ -61,23 +69,38 @@ defmodule Onvif.Device do
     {:ok, device}
   end
 
-  defp get_address(addresses, use_ipv6?) do
-    ipv6_address =
-      Enum.find(addresses, fn address ->
+  defp get_address(addresses, prefer_https?, prefer_ipv6?) do
+    ipv6_addresses =
+      Enum.filter(addresses, fn address ->
         uri = URI.parse(address)
-        String.contains?(uri.host, ":")
+
+        cond do
+          prefer_https? -> String.contains?(uri.host, ":") and uri.scheme == "https"
+          true -> String.contains?(uri.host, ":")
+        end
       end)
 
-    if use_ipv6? and !is_nil(ipv6_address) do
-      ipv6_address
-    else
-      Enum.at(addresses, 0)
+    ipv4_addresses =
+      Enum.filter(addresses, fn address ->
+        uri = URI.parse(address)
+
+        cond do
+          prefer_https? -> String.contains?(uri.host, ".") and uri.scheme == "https"
+          true -> String.contains?(uri.host, ".")
+        end
+      end)
```

Comment:
Could move it to a util file or break it into a another private function if needed. I feel right now we can be verbose and break it up when the functionality grows.